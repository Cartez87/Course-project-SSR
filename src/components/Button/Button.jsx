import React from "react";
import propTypes from 'prop-types';

import './Button.scss';

const Button = ({ children, color, ...attrs }) => {

  const BUTTON_COLOR = {
    PRIMARY: 'btn-primary',
    SECONDARY: 'btn-secondary',
    TRANSPARENT: 'btn-transparent'
  }

  const colorClassName = BUTTON_COLOR[color || 'PRIMARY'];

  return (
    <button 
      className={"btn " + colorClassName}
      {...attrs}
    >
      {children}
    </button>
  )
}

Button.propTypes = {
  children: propTypes.string,
  color: propTypes.string
}

export default Button;