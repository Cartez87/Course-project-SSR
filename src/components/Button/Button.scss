.btn {
  transition: background 0.15s ease-in-out, color 0.15s ease-in-out;
  min-width: 235px;
  text-transform: uppercase;
  border-radius: 4px;
  border: none;
  padding: 0.7em;
  font-size: 20px;
  font-weight: 500;

  &:focus {
    box-shadow: none;
    border-color: none;
  }

  &.btn-primary {
    background: #F65261;
    color: white;

    &:hover {
      background: #c52f3b;
    }
  }

  &.btn-secondary {
    background: rgba(96, 96, 96, 0.68);
    color: #F65261;
    font-weight: 600;
    min-width: 177px;
    padding: 0.4em;

    &:hover {
      background: rgba(96, 96, 96, 1);
      color: white;
    }
  }
  &.btn-transparent {
    border: 1.5px solid #F65261;
    background: none;
    color: #F65261;
    padding: 0.55em;
    min-width: 182px;

    &:hover {
      background: #F65261;
      color: white;
    }
  }
}
