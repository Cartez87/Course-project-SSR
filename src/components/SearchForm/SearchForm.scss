
.search-form {
  max-width: 800px;
  width: 100%;
  position: relative;
  z-index: 2;
}

.search-form input{
  width: 100%;
  padding: 0.7em;
  margin-right: 10px;
  background: rgba(50, 50, 50, 0.8);
  mix-blend-mode: normal;
  border-radius: 4px;
  color: white;
  font-size: 20px;
  border: 0;
}

.search-form .form-control:focus {
  box-shadow: none;
  border-color: #F65261;
}