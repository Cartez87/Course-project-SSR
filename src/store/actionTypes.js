
export const FETCH_MOVIES = 'MOVIECARD/FETCH_MOVIES';
export const FETCH_MOVIES_FAILED = 'MOVIECARD/FETCH_MOVIES_FAILED';
export const FILTER_MOVIES = 'MOVIECARD/FILTER_MOVIES';
export const SORT_MOVIES = 'MOVIECARD/SORT_MOVIES';
export const SELECT_MOVIE = 'MOVIECARD/SELECT_MOVIE';