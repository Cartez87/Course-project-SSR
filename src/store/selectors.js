
export const selectedMovieSelector = state => state.movieReducer.selectedMovie;