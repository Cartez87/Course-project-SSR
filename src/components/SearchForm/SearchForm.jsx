import React, { useState } from 'react';
import propTypes from 'prop-types';
import Button from '../Button';

import {useNavigate} from 'react-router-dom';
import './SearchForm.scss';
// import { useDispatch } from 'react-redux';

const SearchForm = ({ searchMovie }) => {
  const [ query, setQuery ] = useState('');
  // const dispatch = useDispatch();
  // const navigate = useNavigate();
  
  const handleChange = (e) => {
    e.preventDefault();
    setQuery(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    // dispatch(searchMovie(query));
    // navigate(`/search/${query}`);
  }

  return (
    <form onSubmit={handleSubmit} className="search-form d-flex">
      <input 
        onChange={handleChange} 
        value={query}
        type="search" 
        className="form-control"
        aria-label="input"
        placeholder="What do you want to watch?"
      />
      <Button
        type="submit"
      >    
        Search
      </Button>
    </form>
  )
}

SearchForm.propTypes = {
  searchMovie: propTypes.func
}

export default SearchForm;