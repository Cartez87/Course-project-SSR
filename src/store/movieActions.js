import { getSortedFilteredFilms, deleteMovies, getSearchMovies } from "../services/movieService";
import { FETCH_MOVIES, FILTER_MOVIES, SORT_MOVIES, SELECT_MOVIE } from "./actionTypes";

export const fetchMovies = (filter='', sortOrder='') => {
  return async dispatch => {
    const data = await getSortedFilteredFilms(filter, sortOrder);
    console.log(data, 'aaaaa')
    dispatch({type: FETCH_MOVIES, payload: data});
  }
};

export const filterMovies = (filter) => {
  return {
    type: FILTER_MOVIES, 
    payload: filter
  }
};

export const sortMovies = (sortOrder) => {
  return {
    type: SORT_MOVIES, 
    payload: sortOrder
  }
};

export const selectMovie = (id) => {
  return {
    type: SELECT_MOVIE,
    payload: id
  }
};

export const deleteMovie = (id, filter='', sortOrder='') => {
  return (dispatch) => {
    deleteMovies(id)
    .then(() => dispatch(fetchMovies(filter, sortOrder)));
  }
};

export const searchMovie = (search) => {
  return async (dispatch) => {
    const searchData = await getSearchMovies(search);
    console.log(searchData)
    dispatch({type: FETCH_MOVIES, payload: searchData});
  }
};