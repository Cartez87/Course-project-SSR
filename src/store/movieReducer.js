import { FETCH_MOVIES, FILTER_MOVIES, SORT_MOVIES, SELECT_MOVIE } from "./actionTypes"

export const initialState = {
  fetchedMovies: [],
  sortBy: "release_date",
  sort: {
    value: '',
    label: '',
    order: ''
  },
  filter: '',
  selectedMovie: null
}

export default function movieReducer (state = initialState, {type, payload}) {
  switch (type) {
    case FETCH_MOVIES:
      return { ...state, fetchedMovies: payload }
    case FILTER_MOVIES:
      return { ...state, filter: payload }
    case SORT_MOVIES:
      return { ...state, sort: payload }
    case SELECT_MOVIE:
      return { ...state, selectedMovie: payload }
    default: return state
  }
}
