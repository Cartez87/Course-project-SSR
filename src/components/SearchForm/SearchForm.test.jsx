import React from 'react';
import { render as renderRedux, screen, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';

import SearchForm from './SearchForm';
import Button from '../Button';
import store from '../../store/store';
import { handleChange } from './SearchForm';

const render = component => renderRedux(
    <Provider store={store}>
       {component}
    </Provider>
);

describe('Search component', () => {
    let component;
    test('Renders SearchForm without crashing', () => {
        render(<SearchForm />);
    });
    test('Renders SearchForm with snapshot', () => {
        component = render(<SearchForm />);
        expect(component).toMatchSnapshot();
    });
    test('renders SearchForm with placeholder', () => {
        const { getByPlaceholderText } = render(<SearchForm placeholder={'What do you want to watch?'} />);
        const linkText = getByPlaceholderText(/what do you want to watch?/i);
        expect(linkText).toBeTruthy();
    });
    test('renders SearchForm without placeholder', () => {
        const { getByPlaceholderText } = render(<SearchForm />);
        const linkText = getByPlaceholderText(/what do you want to watch?/i);
        expect(linkText).toBeTruthy();
    });
    test('onChange input', () => {
        component = render(<SearchForm onChange={handleChange}/>);
        fireEvent.change(component.getByLabelText('input'), {target: {value: 'react'}});
        expect(component.getByLabelText('input').value).toBe('react');
    });
    test('onClick button', () => {
        const onClick = jest.fn();
        const { getByText } = render(<Button onClick={onClick}>Search</Button>);
        fireEvent.click(getByText(/search/i));
        expect(onClick).toHaveBeenCalled();
    });
});