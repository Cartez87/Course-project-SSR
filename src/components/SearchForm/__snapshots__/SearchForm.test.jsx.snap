// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component Renders SearchForm with snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="search-form d-flex"
      >
        <input
          aria-label="input"
          class="form-control"
          placeholder="What do you want to watch?"
          type="search"
          value=""
        />
        <button
          class="btn btn-primary"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="search-form d-flex"
    >
      <input
        aria-label="input"
        class="form-control"
        placeholder="What do you want to watch?"
        type="search"
        value=""
      />
      <button
        class="btn btn-primary"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
